scripts/a2_collections

```diff
@@ -137,3 +142,97 @@ authLoad() {
+#     This generates dummy data that is not particularly realistic
+#     but can be used to observe general performance characteristics.
+#     NB: When generating policies, it builds one policy per token,
+#     so you should also generate a set of tokens with the same parameters.
+#     This function also allows you to delete your generated resources just as easily.
+# 
+# Example:
+#     Create 500 policies starting with a seed of 101:
+#         $ authGen policies create 500 101
+#     Per above note, those policies need these members:
+#         $ authGen tokens create 500 101
+#     Now, just for illustration, delete the last 100 of those 500 policies:
+#         $ authGen policies delete 100 501
+#
+# Options:
+#    -2 or --v2           use IAM V2 endpoints
+#
+# TODO: Except for common things (e.g. teams) this supports only IAM v1 so far.
+#
+authGen() {
+  # process options
+  auth_path="api/v0/auth"
+  while [[ "$1" =~ ^- ]]; do
+    case "$1" in
+      -2|--v2) auth_path="apis/iam/v2beta"; shift;;
+      -*) echo "'$1' unknown"; return 1;;
+    esac
+  done
+
+  # prerequisites
+  ([ -z "$TOK" ] || [ -z "$TARGET_HOST" ]) && { echo "Need TOK and TARGET_HOST variables"; return; }
+  [ -z "$4" ] && { echo "usage: authGen [ <options> ] <resource> <mode> <count> <seed>"; return; }
+  if ! type jo &> /dev/null
+  then
+    echo "Need jo installed"; return;
+  fi
+  RESOURCE=$1
+  MODE=$2
+  COUNT=$3
+  SEED=$4
+  ID_PREFIX=test-$RESOURCE
+  [[ ! $MODE =~ ^(create|delete)$ ]] && { echo "mode must be 'create' or 'delete'"; return; }
+  [[ ! $RESOURCE =~ ^(tokens|users|teams|policies)$ ]] && { echo "resource must be in: tokens, users, teams, policies"; return; }
+
+  if [[ $MODE == create ]]; then
+    echo "Creating $COUNT $RESOURCE..."
+    for (( i = 0; i < $COUNT; i++ )) 
+    do
+      local id_index=$(($i+$SEED))
+      local id=$ID_PREFIX-$id_index
+      local name="$ID_PREFIX $id_index"
+      local json=""
+      case "$RESOURCE" in
+        tokens) 
+          json=$(jo -p id=$id name="$name" description="test token for $name" active=true) ;;
+        users) 
+          json=$(jo -p id=$id name="$name" username=$id password=chefautomate active=true) ;;
+        teams) 
+          json=$(jo name="$name" description="test team for $name") ;;
+        policies) 
+          json=$(jo subjects=$(jo -a token:test-tokens-$id_index) action=read resource="compliance:*") ;;
+      esac
+      curl -sSkH "api-token: $TOK" $TARGET_HOST/$auth_path/$RESOURCE -X POST --data "$json"
+      echo
+    done
+ 
+  elif [[ $MODE == delete ]]; then
+    echo "Removing $COUNT $RESOURCE..."
+    for (( i = 0; i < $COUNT; i++ )) 
+    do
+      local id_index=$(($i+$SEED))
+      local name="$ID_PREFIX $id_index"
+      local id=$ID_PREFIX-$id_index
+      case "$RESOURCE" in
+        tokens)  ;;
+        users)  ;;
+        teams) 
+          local teams=$(curl -sSkH "api-token: $TOK" "$TARGET_HOST/$auth_path/$RESOURCE")
```

Comment:
This doesn't matter in this script, but we tend to prefer
```suggestion
          local teams
          team=$(curl -sSkH "api-token: $TOK" "$TARGET_HOST/$auth_path/$RESOURCE")
```

Using `local` and a command substitution assignment together means you lose the return value and if you are using `set -e`, you don't get the exit-on-failure behavior you might expect.  See

https://github.com/koalaman/shellcheck/wiki/SC2155

for more info.